### PR TITLE
feat(ci): parallelize e2e test

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

As title

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Comment by skyzh** [comment]
Due to unknown bugs, we use `default` mode for parallel test for now. After the bugs are solved, we can use the normal ci-3cn-1fe mode.

**Comment by skyzh** [comment]
At least we can now fail fast :)

File: ci/scripts/e2e-test-parallel.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+echo "--- e2e, ci-3cn-1fe, delta join"
+~/cargo-make/makers ci-start default
+timeout 3m sqllogictest -p 4566 -d dev './e2e_test/streaming_delta_join/**/*.slt'
+
+echo "--- Kill cluster"
+~/cargo-make/makers ci-kill
+
+echo "--- e2e, ci-3cn-1fe, batch distributed"
+~/cargo-make/makers ci-start default
+timeout 2m sqllogictest -p 4566 -d dev './e2e_test/ddl/**/*.slt' -j 8
```
Comment (BugenZhao) [inline ci/scripts/e2e-test-parallel.sh:58]: There're some tests related to the catalog like `SHOW DATABASES` in `ddl`. Seems we cannot run them in parallel?

Comment (skyzh) [reply ci/scripts/e2e-test-parallel.sh:58]: Oops, that's true.

**Review by fuyufjh — APPROVED** [review]
LGTM

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**